psi4/driver/driver_util.py

```diff
@@ -282,50 +252,102 @@ def parse_cotton_irreps(irrep: Union[str, int], point_group: str) -> int:
     raise ValidationError(f"""Irrep '{irrep}' not valid for point group '{point_group}'.""")
 
 
-def negotiate_derivative_type(ptype: str,
-                              method: str,
-                              user_dertype: Optional[int],
-                              verbose: int = 1,
-                              return_strategy: bool = False,
-                              proc: Optional[Dict] = None) -> Union[int, str]:
+def negotiate_derivative_type(
+    target_dertype: str,
+    method: str,
+    user_dertype: Optional[int],
+    verbose: int = 1,
+    proc: Optional[Dict] = None,
+) -> Union[Tuple[int, int], Tuple[str, str]]:
     r"""Find the best derivative level (0, 1, 2) and strategy (analytic, finite difference)
-    for `method` to achieve `ptype` within constraints of `user_dertype`.
+    for `method` to achieve `target_dertype` within constraints of `user_dertype`.
 
     Parameters
     ----------
-    ptype : {'energy', 'gradient', 'hessian'}
+    target_dertype: {"energy", "gradient", "hessian", "properties"}
```

Comment:
In `sort_derivative_type`, `properties` does not count as a dertype. Here it does. Is it possible/advisable to get these to agree on whether `properties` is a dertype or not?

Reply:
Not really, at least right now. The trio of functions is really suited for negotiating e/g/h; "properties" was forcibly inserted a couple weeks ago so that `properties("ccsd/cc-pvdz")` (slash invokes cbs machinery) could run. So that's why the top-level fn `negotiate_derivative_type` acknowledges properties, while the bottom-level fn `sort_derivative_type`, which doesn't get called on the properties route, doesn't deal in properties. I think the whole driver would have to be more fine-grained in how it offers what properties are available per method rather than the on/off of `procedures["properties"][method]` to justify tighter integration of properties into these probing functions. It could be nice in future, but for now I'm content with this basic working integration.